users/tests/test_unit_users_apiview.py

```diff
@@ -12,26 +12,29 @@
 class UserListTests(TestCase):
     def setUp(self):
         self.user = CustomUser(
-            email="testuser@codepoets.it", password="newuserpasswd", first_name="John", last_name="Doe", country="PL"
+            email="testuser@codepoets.it",
+            password="newuserpasswd",
+            first_name="John",
+            last_name="Doe",
+            country="PL",
+            user_type=CustomUser.UserType.ADMIN.name,
         )
         self.user.full_clean()
         self.user.save()
         self.url = reverse("custom-users-list")
 
     def test_user_list_view_should_display_users_list_on_get(self):
-        request = APIRequestFactory().get(path=self.url)
-        request.user = self.user
-        response = views.UserList.as_view()(request)
+        self.client.force_login(self.user)
+        response = self.client.get(self.url)
         self.assertEqual(response.status_code, 200)
         self.assertContains(response, self.user.user_type)
-        users_list = response.data["users_list"]
-        self.assertTrue(self.user in users_list)
+        self.assertContains(response, self.user.email)
+        self.assertContains(response, self.user.first_name)
+        self.assertContains(response, self.user.last_name)
 
     def test_user_list_view_should_not_be_accessible_for_unauthenticated_user(self):
```

Comment:
How about adding a test that non-admin users can't access the view? There is already similar one called `test_delete_project_function_view_should_not_delete_project_on_non_admin_request`.